src/cpp/b2w.cpp

```diff
@@ -130,8 +136,8 @@ int main(int argc, char* argv[])
                 exit(EXIT_FAILURE);
         }
     }
-    if (argc < 11 || argc > 12) {
-        std::fprintf(stderr, "%s", help_string);
+    if ((argc - optind) < 2 or (argc - optind) > 3) {
+        std::fprintf(stderr, "%d parameters left, %s", argc - optind, help_string);
```

Comment:
(not for this PR) could you maybe look into getting rid of all the "C-in-C++"-isms. Like using `std::cerr` over C-style `fprintf` stuff?